docs/database/oracle-entity-framework-integration.md

```diff
@@ -1,125 +1,224 @@
+
+## Client integration
+
+You need an Oracle database and connection string for accessing the database. To get started with the .NET Aspire Oracle client integration, install the [📦 Aspire.Oracle.EntityFrameworkCore](https://www.nuget.org/packages/Aspire.Microsoft.Data.SqlClient) NuGet package in the client-consuming project, that is, the project for the application that uses the Oracle client. The Oracle client integration registers a <xref:System.Data.Entity.DbContext> instance that you can use to interact with Oracle.
 
 ### [.NET CLI](#tab/dotnet-cli)
 
 ```dotnetcli
-dotnet add package Aspire.Hosting.Oracle
+dotnet add package Aspire.Oracle.EntityFrameworkCore
 ```
 
 ### [PackageReference](#tab/package-reference)
 
 ```xml
-<PackageReference Include="Aspire.Hosting.Oracle"
+<PackageReference Include="Aspire.Oracle.EntityFrameworkCore"
                   Version="*" />
 ```
 
 ---
 
-In your app host project, register an Oracle container and consume the connection using the following methods:
+### Add Oracle client
+
+In the _:::no-loc text="Program.cs":::_ file of your client-consuming project, call the <xref:Microsoft.Extensions.Hosting.AspireOracleEFCoreExtensions.AddOracleDatabaseDbContext*> extension method on any <xref:Microsoft.Extensions.Hosting.IHostApplicationBuilder> to register a `DbContext` for use via the dependency injection container. The method takes a connection name parameter.
 
 ```csharp
-var builder = DistributedApplication.CreateBuilder(args);
+builder.AddOracleDatabaseDbContext<ExampleDbContext>(connectionName: "oracledb");
+```
 
-var oracle = builder.AddOracle("oracle");
-var oracledb = oracle.AddDatabase("oracledb");
+> [!TIP]
+> The `connectionName` parameter must match the name used when adding the Oracle database resource in the app host project. In other words, when you call `AddDatabase` and provide a name of `oracledb` that same name should be used when calling `AddOracleDatabaseDbContext`. For more information, see [Add Oracle server resource and database resource](#add-oracle-server-resource-and-database-resource).
 
-var myService = builder.AddProject<Projects.MyService>()
-                       .WithReference(oracledb);
+You can then retrieve the <xref:Microsoft.EntityFrameworkCore.DbContext> instance using dependency injection. For example, to retrieve the connection from an example service:
+
+```csharp
+public class ExampleService(ExampleDbContext context)
+{
+    // Use database context...
+}
 ```
 
-When you want to explicitly provide a password, you can provide it as a parameter. Consider the following alternative example:
+For more information on dependency injection, see [.NET dependency injection](/dotnet/core/extensions/dependency-injection).
 
-```csharp
-var password = builder.AddParameter("password", secret: true);
+### Add Oracle database context with enrichment
```

Comment:
Is there a keyed API, for example `Add keyed Oracle client` section would be missing if there is.